lib/opencensus/trace/message_event.rb

```diff
@@ -0,0 +1,78 @@
+# Copyright 2017 OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module OpenCensus
+  module Trace
+    ##
+    # An event describing a message sent/received between Spans.
+    #
+    class MessageEvent < TimeEvent
+      # Unknown event type.
+      MESSAGE_EVENT_TYPE_UNSPECIFIED  = :TYPE_UNSPECIFIED
+
+      # Indicates a sent message.
+      MESSAGE_EVENT_TYPE_SENT         = :SENT
+
+      # Indicates a received message.
+      MESSAGE_EVENT_TYPE_RECEIVED     = :RECEIVED
+
+      ##
+      # The type of MessageEvent. Indicates whether the message was sent or
+      # received.
```

Comment:
Same comment as earlier. Document that you should use the constants provided in this class.